Assets/HoloToolkit/Input/Scripts/Utilities/MicrophoneHelper.cs

```diff
@@ -0,0 +1,145 @@
+﻿
+//*********************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+//
+// This sample code is a modified version of an official UWP sample which can be found on:
+// https://github.com/Microsoft/Windows-universal-samples/blob/master/Samples/SpeechRecognitionAndSynthesis/cs/AudioCapturePermissions.cs
+//
+//*********************************************************
+
+#if ENABLE_WINMD_SUPPORT
+using System;
+using System.Threading.Tasks;
+using Windows.Media.Capture;
+#endif
+
+namespace HoloToolkit.Unity
+{
+    enum MicrophoneStatus
+    {
+        MicrophoneReady,
+        MicrophoneUseNotAuthorized,
+        MicrophoneNotPresent,
+        MicrophoneMuted,
+        MicrophoneUseFailed,
+        MediaComponentsMissing,
+    }
+
+    /// <summary>
+    /// Provides a working status information of the default microphone on the system.
+    /// Class is using UWP APIs. Everything UWP related is wrapped around ENABLE_WINMD_SUPPORT
+    /// preprocessor definition. The definition is enabled in Unity in the following 2 cases:
+    ///     - IL2CPP backend and .NET 4.6
+    ///     - .NET backend and Net Core compilation override
+    /// For full details see unity documentation page
+    /// https://docs.unity3d.com/Manual/IL2CPP-WindowsRuntimeSupport.html
+    /// </summary>
+    class MicrophoneHelper
+    {
+#if ENABLE_WINMD_SUPPORT
+        // If no recording device is attached, attempting to get access to audio capture devices will throw 
+        // a System.Exception object, with HResult = ‭0xC00DABE0‬ set.
```

Comment:
nit: `<summary>` tags?

Reply:
I'm not sure what you mean here. Can you elaborate?

Reply:
Update the comment to use `<summary>` tags.